metrics_api/lib/opentelemetry/metrics/meter.rb

```diff
@@ -69,6 +69,22 @@ def create_instrument(kind, name, unit, description, callback)
         end
       end
 
+      def invalid_name?(name)
+        !NAME_REGEX.match?(name)
+      end
```

Comment:
I noticed that `!NAME_REGEX.match?(name)` already accounts for `name.nil?` and `name.empty?` so I've removed them and kept only the regex match.